marklogic-data-hub-central/ui/src/components/common/hc-popover-search/hc-popover-search.tsx

```diff
@@ -0,0 +1,174 @@
+import {HCButton, HCInput} from "@components/common";
+import {Search} from "react-bootstrap-icons";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faSearch} from "@fortawesome/free-solid-svg-icons";
+import styles from "./hc-popover-search.module.scss";
+
+interface Props {
+  // input props
+  inputValue: string,
+  inputId?: string,
+  inputAriaLabel?: string,
+  inputPlaceholder?: string,
+
+  // dialog props
+  popoverId?: string,
+  popoverAriaLabel?: string,
+  closeOnClickOutside?: boolean
+
+  // icon props
+  searchIconId?: string,
+  searchIconAriaLabel?: string,
+
+  // button search props
+  searchButtonText?: string,
+  searchButtonId?: string,
+  onSearch: (value: string) => void,
+
+  // button reset props
+  resetButtonText?: string,
+  resetButtonId?: string,
+  onReset: () => void,
+}
+
+const HCPopoverSearch: React.FC<Props> = ({
+  inputValue, inputId, inputAriaLabel, inputPlaceholder,
+  popoverId, popoverAriaLabel, closeOnClickOutside,
+  searchIconId, searchIconAriaLabel,
+  searchButtonText, searchButtonId, onSearch,
+  resetButtonText, resetButtonId, onReset
+}: Props) => {
+  const [searchText, setSearchText] = useState(inputValue);
+  const [isVisible, setIsVisible] = useState(false);
+  const [isEventValid, setIsEventValid] = useState(false);
+  const popoverBodyRef: any = useRef();
+
+  const toggleVisibility = () => {
+    setIsEventValid(prev => true);
+    setIsVisible(isVisible => !isVisible);
+  };
+
+  const handleOuterClick = useCallback(
+    e => {
+      if (popoverBodyRef.current && !popoverBodyRef.current.contains(e.target) && closeOnClickOutside) {
+        // Clicked outside the popover
+        hideDialog();
+      }
+    }, []);
+
+  useEffect(() => {
+    if (isEventValid) {
+      document.addEventListener("click", handleOuterClick);
+    }
+
+    return () => {
+      document.removeEventListener("click", handleOuterClick);
+    };
+  });
+
+  const handleOnChange = ({target: {value}}) => {
+    setSearchText(value);
+  };
+
+  const hideDialog = () => {
+    setIsEventValid(false);
+    setIsVisible(false);
+  };
+
+  const handleOnReset = (event) => {
+    event.stopPropagation();
+    hideDialog();
+    setSearchText("");
+    onReset();
+  };
+
+  const handleOnSearch = (event) => {
+    event.stopPropagation();
+    hideDialog();
+    onSearch(searchText);
+  };
+
+  return (
+    <div onClick={e => e.stopPropagation()} className={"position-relative d-inline-block"}>
+      <span>
+        <FontAwesomeIcon
+          className={isVisible || searchText ? styles.filterIconActive : styles.filterIcon}
+          id={searchIconId}
+          data-testid={searchIconId}
+          aria-label={searchIconAriaLabel || searchIconId}
+          icon={faSearch}
+          size="lg"
+          onClick={() => toggleVisibility()}
+        />
+      </span>
+      {isVisible &&
+        <div
+          id={popoverId}
+          ref={popoverBodyRef}
+          data-testid={popoverId}
```

Comment:
If we are setting aria-label here then better to avoid data-testId for any new component now. The end goal here is to make the new component accessible (to testing frameworks, screen readers etc.), for which aria-label or Id attributes should be preferred.
We should avoid using custom data Ids like data-testId, unless a situation calls for it.

Reply:
done